.github/workflows/codeql.yml

```diff
@@ -20,24 +21,37 @@ jobs:
     # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
     runs-on: ubuntu-latest
 
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
 
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go.mod
```

Comment:
We need to setup Go env to avoid the following error:

>  go mod tidy -go 1.19
  invalid value "1.19" for flag -go: maximum supported Go version is 1.18
  usage: go mod tidy [-e] [-v] [-go=version] [-compat=version]
  Run 'go help mod tidy' for details.
  make: *** [Makefile:107: tidy] Error 2

https://github.com/kubeflow/mpi-operator/actions/runs/4239854555/jobs/7368287513#step:4:134